server/controllers/authController.js

```diff
@@ -0,0 +1,58 @@
+import passport from 'passport';
+import jwt from 'jsonwebtoken';
+
+const { JWT_EXPIRES_MS, JWT_SECRET } = process.env;
+/**
+ * This class contains all methods
+ * required to handle all
+ * authentication routes.
+ */
+class AuthController {
+  /**
+     * This method handle the sign request.
+     * @param {object} req The user's request.
+     * @param {object} res The response.
+     * @returns {object} The status and some data of the user.
+     */
+  static signIn(req, res) {
+    passport.authenticate(
+      'local',
+      { session: false },
+      (error, user) => {
+        if (error || !user) {
+          res.status(404).json({ error });
+        }
+
+        const { id, email, lastname } = user;
+        /** This is what goes in our JWT */
+        const payload = {
+          id,
+          expires: Date.now() + parseInt(JWT_EXPIRES_MS, 4),
+        };
+
+        /** assigns payload to req.user */
+        req.login(payload, { session: false }, (err) => {
+          if (err) {
+            res.status(400).send({ err });
+          }
+
+          /** generate a signed json web token and return it in the response */
+          const token = jwt.sign(JSON.stringify(payload), JWT_SECRET);
+
+          res.cookie('jwt', jwt, { httpOnly: true, secure: true });
+          return res.status(200).send({
+            status: 200,
+            message: 'User signed in successfully',
+            data: {
+              token,
+              lastname,
+              email
+            }
+          });
+        });
+      },
+    )(req, res);
+  }
```

Comment:
what is the use of this ?

Reply:
Every time an API request is made, the request (req) passes through the authentication strategy used by passport, and if successfully authenticated, then the passport binds the details of the user to the request(req).